addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -4,11 +4,25 @@ odoo.define('website.s_table_of_content_options', function (require) {
 const options = require('web_editor.snippets.options');
 
 options.registry.TableOfContent = options.Class.extend({
+    /**
+     * @override
+     */
+    init() {
+        this.targetedElements = "h1, h2";
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    onBuilt() {
+        this._resetIds();
+        this._generateNav();
+        return this._super(...arguments);
```

Comment:
Nitpicking: In general we don't call `_super` in those `onXXX` method (as they are designed to be empty in the parent implementation).